src/main/kotlin/nmcp/NmcpAggregationExtension.kt

```diff
@@ -0,0 +1,71 @@
+package nmcp
+
+import gratatouille.GExtension
+import nmcp.internal.configureAttributes
+import nmcp.internal.nmcpConsumerConfigurationName
+import nmcp.internal.task.registerPublishTask
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.tasks.bundling.Zip
+
+@GExtension(pluginId = "com.gradleup.nmcp.aggregation")
+open class NmcpAggregationExtension(private val project: Project) {
+    internal val spec = project.objects.newInstance(NmcpSpec::class.java)
+
+    internal val consumerConfiguration = project.configurations.create(nmcpConsumerConfigurationName) {
+        it.isCanBeResolved = true
+        it.isCanBeConsumed = false
+
+        it.configureAttributes(project)
+    }
+
+    val zipTaskProvider = project.tasks.register("zipAggregation", Zip::class.java) {
+        it.archiveFileName.set("aggregation.zip")
+        it.destinationDirectory.set(project.layout.buildDirectory.dir("nmcp/zip"))
+        it.from(consumerConfiguration.elements.map {
+            check (it.isNotEmpty()) {
+                "nmcp: aggregation is empty. Make sure to create publications in sub-projects and specify projects to publish by adding them to the 'nmcpAggregation' configuration."
+            }
+            it.map {
+                project.zipTree(it)
```

Comment:
This looks like a "project usage at the execution time", and it should probably be replaced with [`ArchiveOperations.zipTree`](https://docs.gradle.org/current/javadoc/org/gradle/api/file/ArchiveOperations.html#zipTree(java.lang.Object)) or something like that (e.g. `FileOperations.zipTree`)

Reply:
Good idea. Main reason I haven't done so is that I'm not overly familiar with the artifacts API and I've always used it with files. I've read somewhere that KGP now exposes directories for class files or something like this and it saved a bunch of time when compiling multi modules repositories.

Reply:
Mainly, you expose directory with artifact and add `builtBy` so Gradle recognizes task dependency.
It is virtually the same as files.

Here's an example: https://github.com/sigstore/sigstore-java/blob/75fd1ccbd252c5ea96c16cc00fba522184bffccf/build-logic/publishing/src/main/kotlin/build-logic.publish-to-tmp-maven-repo.gradle.kts#L30-L31

Reply:
For now, I've just [fixed configuration cache](https://github.com/GradleUp/nmcp/pull/29/commits/de6e2e54a79f6aecc3d69675efe88c0fcdb06ab7) 
I've opened a follow up issue to optimize the zip/unzip dance [there](https://github.com/GradleUp/nmcp/issues/33)